include/alpaka/core/BoostPredef.hpp

```diff
@@ -30,12 +30,50 @@
     #define BOOST_PREDEF_MAKE_10_VVRRP(V) BOOST_VERSION_NUMBER(((V)/1000)%100,((V)/10)%100,(V)%10)
 #endif
 
+//---------------------------------------HIP-----------------------------------
+// __HIPCC__ is defined by hipcc (if either __HCC__ or __CUDACC__ is defined)
+#if !defined(BOOST_LANG_HIP)
+  #if defined(__HIPCC__) && ( defined(__CUDACC__) || defined(__HCC__) )
+    #include <hip/hip_runtime.h>
+    //HIP defines "abort()" as "{asm("trap;");}", which breaks some kernels
+    #undef abort
+    // there is no HIP_VERSION macro
+    #define BOOST_LANG_HIP BOOST_VERSION_NUMBER_AVAILABLE
+  #else
+    #define BOOST_LANG_HIP BOOST_VERSION_NUMBER_NOT_AVAILABLE
+  #endif
+#endif
+
+//-----------------------------------------------------------------------------
+// HSA device architecture detection (HSA generated via HIP(HCC) or HCC directly)
+#if !defined(BOOST_ARCH_HSA)
+    #if defined(__HIP_DEVICE_COMPILE__) && __HIP_DEVICE_COMPILE__==1 && defined(__HCC__) \
+        || (defined(__HCC_ACCELERATOR__) && __HCC_ACCELERATOR__!=0)
+        // __HIP_DEVICE_COMPILE__ does not represent feature capability of target device like CUDA_ARCH.
+        // For feature detection there are special macros, see ROCm's HIP porting guide.
+        #define BOOST_ARCH_HSA BOOST_VERSION_NUMBER_AVAILABLE
+    #else
+        #define BOOST_ARCH_HSA BOOST_VERSION_NUMBER_NOT_AVAILABLE
+    #endif
+#endif
+
+//-----------------------------------------------------------------------------
+// hcc HSA compiler detection
+#if !defined(BOOST_COMP_HCC)
+    #if defined(__HCC__)
+        #define BOOST_COMP_HCC BOOST_VERSION_NUMBER_AVAILABLE
+    #else
+        #define BOOST_COMP_HCC BOOST_VERSION_NUMBER_NOT_AVAILABLE
+    #endif
+#endif
+
 //-----------------------------------------------------------------------------
 // In boost since 1.68.0
 // CUDA language detection
 // - clang defines __CUDA__ and __CUDACC__ when compiling CUDA code ('-x cuda')
 // - nvcc defines __CUDACC__ when compiling CUDA code
-#if !defined(BOOST_LANG_CUDA)
+#if !defined(BOOST_LANG_CUDA) && \
+    !(BOOST_LANG_HIP && defined(__HIPCC__))
```

Comment:
We would have to add this to boost as well or is there another way to fix this behaviour on our side? Maybe undef `BOOST_LANG_CUDA` afterwards when `BOOST_LANG_HIP` is detected? I wanted to remove this whole block as soon as we increase our required boost version.

Reply:
Could you please implement the undef solution after for now? I would like to keep the implementation here exactly the one that is within boost. Furthermore, if `BOOST_LANG_CUDA` is defined by boost itself, we still have both languages defined.

Reply:
ah ok, yes sure.

Reply:
I'll remove that block above, and insert an undef after `BOOST_LANG_HIP` is defined:
```C++
// ...
#define BOOST_LANG_HIP BOOST_VERSION_NUMBER_AVAILABLE
#if defined(BOOST_LANG_CUDA) && BOOST_LANG_CUDA      
    #undef BOOST_LANG_CUDA                           
#endif                                               
// ...
```

Reply:
or I also could set it to `BOOST_VERSION_NUMBER_NOT_AVAILABLE` ?

Reply:
I think `BOOST_VERSION_NUMBER_NOT_AVAILABLE ` would be correct.